Block/BlockCharcoalPile.cs

```diff
@@ -1,9 +1,66 @@
 ﻿using Vintagestory.API.Common;
+using Vintagestory.API.Datastructures;
+using Vintagestory.API.MathTools;
 
 namespace Vintagestory.GameContent
 {
     public class BlockCharcoalPile : BlockLayeredSlowDig
     {
+        public override bool OnFallOnto(IWorldAccessor world, BlockPos pos, Block block, TreeAttribute blockEntityAttributes)
+        {
+            Block nBlock = (BlockCharcoalPile)world.BlockAccessor.GetMostSolidBlock(pos);
+            if (block is BlockCharcoalPile && nBlock is BlockCharcoalPile)
+            {
+
+                Block uBlock = block;
+                while (((BlockCharcoalPile)nBlock).CountLayers() < 8 && uBlock != null)
+                {
+                    nBlock = ((BlockCharcoalPile)nBlock).GetNextLayer(world);
+                    uBlock = ((BlockCharcoalPile)uBlock).GetPrevLayer(world);
+                }
+
+                world.BlockAccessor.SetBlock(nBlock.BlockId, pos);
+                world.BlockAccessor.TriggerNeighbourBlockUpdate(pos);
+                if (uBlock != null)
+                {
+                    world.BlockAccessor.SetBlock(uBlock.BlockId, pos.UpCopy());
+                    world.BlockAccessor.TriggerNeighbourBlockUpdate(pos.UpCopy());
+                }
+                return true;
+            }
+
+            return base.OnFallOnto(world, pos, block, blockEntityAttributes);
+        }

```

Comment:
Allowing for falling charcoal piles to merge into the piles they land on when given the gravity behavior.